File: java/rna-transcription/src/main/java/RnaTranscription.java
class RnaTranscription {

    String transcribe(String dnaStrand) {

        var sb = new StringBuilder();

        for (var c : dnaStrand.toCharArray()) {

            switch (c) {
                case 'G':
                    sb.append('C');
                    break;
                case 'C':
                    sb.append('G');
                    break;
                case 'T':
                    sb.append('A');
                    break;
                case 'A':
                    sb.append('U');
                    break;
                // TODO: maybe there should be a default case
            }
        }

        return sb.toString();
    }
}

File: csharp/rna-transcription/RnaTranscription.cs
using System;
using System.Text;

// There is probably a much nicer way to write this
public static class RnaTranscription
{
    public static string ToRna(string nucleotide)
    {
        var sb = new StringBuilder();

        foreach (char c in nucleotide)
        {
            switch (c)
            {
                case 'G':
                    sb.Append('C');
                    break;
                case 'C':
                    sb.Append('G');
                    break;
                case 'T':
                    sb.Append('A');
                    break;
                case 'A':
                    sb.Append('U');
                    break;
            }
        }

        return sb.ToString();
    }
}